pkg/resources/resources_manager_test.go

```diff
@@ -0,0 +1,295 @@
+             "rdmaHcaMax": 1000,
+             "devices": ["ib0", "ib1"]
+           }}},
+        ]}`
+			fs := &utils.FakeFilesystem{
+				Dirs: []string{"tmp"},
+				Files: map[string][]byte{
+					"tmp/config.json": []byte(configData),
+				},
+			}
+			defer fs.Use()()
+
+			rm := &resourceManager{configFile: fs.RootDir + "/tmp/config.json"}
+			err := rm.ReadConfig()
+			Expect(err).To(HaveOccurred())
+		})
+	})
+	Context("ValidateConfigs", func() {
+		It("Valid config list", func() {
+			var configlist []*types.UserConfig
+			rm := &resourceManager{}
+
+			configlist = append(configlist, &types.UserConfig{
+				ResourceName: "test_config",
+				RdmaHcaMax:   100,
+				Devices:      []string{"ib0"}})
+
+			rm.configList = configlist
+			err := rm.ValidateConfigs()
+			Expect(err).ToNot(HaveOccurred())
+		})
+		It("Validate empty config list", func() {
+			rm := &resourceManager{}
+			err := rm.ValidateConfigs()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("no resources configuration found"))
+		})
+		It("resources with invalid name", func() {
+			var configlist []*types.UserConfig
+			rm := &resourceManager{}
+
+			configlist = append(configlist, &types.UserConfig{
+				ResourceName: "test_config$$",
+				RdmaHcaMax:   100,
+				Devices:      []string{"ib0"}})
+
+			rm.configList = configlist
+			err := rm.ValidateConfigs()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("error: resource name \"test_config$$\" contains invalid characters"))
+		})
+		It("resources with repeated names", func() {
+			var configlist []*types.UserConfig
+			rm := &resourceManager{}
+
+			configlist = append(configlist, &types.UserConfig{
+				ResourceName: "test_config",
+				RdmaHcaMax:   100,
+				Devices:      []string{"ib0"}})
+			configlist = append(configlist, &types.UserConfig{
+				ResourceName: "test_config",
+				RdmaHcaMax:   500,
+				Devices:      []string{"ib1"}})
+
+			rm.configList = configlist
+			err := rm.ValidateConfigs()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("error: resource name \"test_config\" already exists"))
+		})
+		It("resources with invalid number of resources", func() {
+			var configlist []*types.UserConfig
+			rm := &resourceManager{}
+
+			configlist = append(configlist, &types.UserConfig{
+				ResourceName: "test_config",
+				RdmaHcaMax:   -100,
+				Devices:      []string{"ib0"}})
+
+			rm.configList = configlist
+			err := rm.ValidateConfigs()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("error: Invalid value for rdmaHcaMax < 0: -100"))
+		})
+		It("resources with no devices", func() {
+			var configlist []*types.UserConfig
+			rm := &resourceManager{}
+
+			configlist = append(configlist, &types.UserConfig{
+				ResourceName: "test_config",
+				RdmaHcaMax:   100})
+
+			rm.configList = configlist
+			err := rm.ValidateConfigs()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("error: no devices provided"))
```

Comment:
I think you need a test for config where device was specified, but is not exist. Also we should validate that the device is RDMA

Reply:
for `ValidateConfigs` the first 2 cases specify devices, also updated the InitServers it is the first case